### PR TITLE
Add csv log option and fix value returned after running only one test

### PR DESCRIPTION
**Pull request by Sae86**

### CODE REVIEW

**Review by dscott90 — APPROVED** [review]

File: chipsec/logger.py [hunk 1]
```diff
@@ -262,6 +265,16 @@ def set_autolog_file(self):
         else:
             print('Unable to autolog')
 
+    def log_csv(self, file_name, test_cases):
+        fields = ['name', 'result', 'code', 'output']
+        if '.csv' not in file_name:
```
Comment (npmitche) [inline chipsec/logger.py:270]: should probably be `if not file_name.endswith('.csv'):` That will make sure the .csv is at the end of the string.

File: chipsec/logger.py [hunk 2]
```diff
@@ -262,6 +265,16 @@ def set_autolog_file(self):
...
+            file_name = file_name + '.csv'
+        with open(file_name, 'w') as csv_file:
+            results_csv = csv.DictWriter(csv_file, fieldnames=fields)
+            results_csv.writeheader()
+            for test_case in test_cases:
+                results_csv.writerow(test_case.get_fields())
```
Comment (npmitche) [inline chipsec/logger.py:276]: If you wanted to omit the "output" field, you could do: `results_csv.writerow(test_case.get_fields()[:3])` I'll leave it up to you to decide if you want to keep it or night. Doesn't make a big difference.

**Review by npmitche — APPROVED** [review]

File: chipsec/logger.py (hunk 2 above)
Comment (Sae86) [reply chipsec/logger.py:276]: Doesn't work with dictionaries :(

**Review by npmitche — CHANGES_REQUESTED** [review]
Need to look at unit test failures.

**Review by npmitche — APPROVED** [review]

**Pull request merged**